File: fluentcore/src/main/java/com/mcreater/fxfluent/util/BrushUtil.java
package com.mcreater.fxfluent.util;

import com.mcreater.fxfluent.brush.AbstractColorBrush;
import javafx.geometry.Insets;
import javafx.scene.control.Labeled;
import javafx.scene.layout.*;
import javafx.scene.paint.Color;

import java.util.List;
import java.util.Vector;
import java.util.function.BiConsumer;

public class BrushUtil {
    public enum BorderOrientation {
        TOP, BOTTOM, LEFT, RIGHT
    }

    public static BiConsumer<Region, AbstractColorBrush> backgroundFill(CornerRadii cornerRadii) {
        return (region, color) -> region.setBackground(new Background(new BackgroundFill(
                color.getPaint(),
                cornerRadii,
                Insets.EMPTY
        )));
    }

    public static BiConsumer<Region, AbstractColorBrush> textFill() {
        return (region, color) -> {
            if (region instanceof Labeled) ((Labeled) region).setTextFill(color.getPaint());
        };
    }
    public static BiConsumer<Region, AbstractColorBrush> borderFill(BorderOrientation orientation, CornerRadii cornerRadii) {
        return (region, color) -> {
            Insets insets = Insets.EMPTY;
            if (region.getBorder() == null || region.getBorder().getStrokes().size() == 0) {
                region.setBorder(new Border(new BorderStroke(
                        orientation == null || orientation == BorderOrientation.TOP ? color.getPaint() : Color.TRANSPARENT,
                        orientation == null || orientation == BorderOrientation.RIGHT ? color.getPaint() : Color.TRANSPARENT,
                        orientation == null || orientation == BorderOrientation.BOTTOM ? color.getPaint() : Color.TRANSPARENT,
                        orientation == null || orientation == BorderOrientation.LEFT ? color.getPaint() : Color.TRANSPARENT,
                        BorderStrokeStyle.SOLID,
                        BorderStrokeStyle.SOLID,
                        BorderStrokeStyle.SOLID,
                        BorderStrokeStyle.SOLID,
                        cornerRadii,
                        new BorderWidths(1),
                        insets
                )));
            }
            else {
                List<BorderStroke> strokes = new Vector<>();
                region.getBorder().getStrokes().forEach(borderStroke -> strokes.add(new BorderStroke(
                        orientation == null || orientation == BorderOrientation.TOP ? color.getPaint() : borderStroke.getTopStroke(),
                        orientation == null || orientation == BorderOrientation.RIGHT ? color.getPaint() : borderStroke.getRightStroke(),
                        orientation == null || orientation == BorderOrientation.BOTTOM ? color.getPaint() : borderStroke.getBottomStroke(),
                        orientation == null || orientation == BorderOrientation.LEFT ? color.getPaint() : borderStroke.getLeftStroke(),
                        BorderStrokeStyle.SOLID,
                        BorderStrokeStyle.SOLID,
                        BorderStrokeStyle.SOLID,
                        BorderStrokeStyle.SOLID,
                        cornerRadii,
                        new BorderWidths(1),
                        insets
                )));
                region.setBorder(new Border(strokes, new Vector<>()));
            }
        };
    }
}

File: fluentcore/src/main/java/com/mcreater/fxfluent/util/BrushUtil.kt
package com.mcreater.fxfluent.util

import com.mcreater.fxfluent.brush.AbstractColorBrush
import javafx.geometry.Insets
import javafx.scene.control.Labeled
import javafx.scene.layout.*
import javafx.scene.paint.Color
import java.util.*
import java.util.function.BiConsumer
import java.util.function.Consumer

class BrushUtil {
    enum class BorderOrientation {
        TOP,
        BOTTOM,
        LEFT,
        RIGHT
    }
    companion object {
        @JvmStatic
        fun backgroundFill(cornerRadii: CornerRadii?): BiConsumer<Region, AbstractColorBrush> {
            return BiConsumer { region: Region, color: AbstractColorBrush ->
                region.background = Background(
                    BackgroundFill(
                        color.getPaint(),
                        cornerRadii,
                        Insets.EMPTY
                    )
                )
            }
        }

        @JvmStatic
        fun textFill(): BiConsumer<Region, AbstractColorBrush> {
            return BiConsumer { region: Region?, color: AbstractColorBrush ->
                if (region is Labeled) region.textFill = color.getPaint()
            }
        }

        @JvmStatic
        fun borderFill(
            orientation: BorderOrientation?,
            cornerRadii: CornerRadii?
        ): BiConsumer<Region, AbstractColorBrush> {
            return BiConsumer { region: Region, color: AbstractColorBrush ->
                val insets = Insets.EMPTY
                if (region.border == null || region.border.strokes.size == 0) {
                    region.border = Border(
                        BorderStroke(
                            if (orientation == null || orientation == BorderOrientation.TOP) color.getPaint() else Color.TRANSPARENT,
                            if (orientation == null || orientation == BorderOrientation.RIGHT) color.getPaint() else Color.TRANSPARENT,
                            if (orientation == null || orientation == BorderOrientation.BOTTOM) color.getPaint() else Color.TRANSPARENT,
                            if (orientation == null || orientation == BorderOrientation.LEFT) color.getPaint() else Color.TRANSPARENT,
                            BorderStrokeStyle.SOLID,
                            BorderStrokeStyle.SOLID,
                            BorderStrokeStyle.SOLID,
                            BorderStrokeStyle.SOLID,
                            cornerRadii,
                            BorderWidths(1.0),
                            insets
                        )
                    )
                } else {
                    val strokes: MutableList<BorderStroke> =
                        Vector()
                    region.border.strokes
                        .forEach(Consumer { borderStroke: BorderStroke ->
                            strokes.add(
                                BorderStroke(
                                    if (orientation == null || orientation == BorderOrientation.TOP) color.getPaint() else borderStroke.topStroke,
                                    if (orientation == null || orientation == BorderOrientation.RIGHT) color.getPaint() else borderStroke.rightStroke,
                                    if (orientation == null || orientation == BorderOrientation.BOTTOM) color.getPaint() else borderStroke.bottomStroke,
                                    if (orientation == null || orientation == BorderOrientation.LEFT) color.getPaint() else borderStroke.leftStroke,
                                    BorderStrokeStyle.SOLID,
                                    BorderStrokeStyle.SOLID,
                                    BorderStrokeStyle.SOLID,
                                    BorderStrokeStyle.SOLID,
                                    cornerRadii,
                                    BorderWidths(1.0),
                                    insets
                                )
                            )
                        })
                    region.border = Border(strokes, Vector())
                }
            }
        }
    }
}